lib/cli/run-helpers.js

```diff
@@ -0,0 +1,335 @@
+'use strict';
+
+const fs = require('fs');
+const debug = require('debug')('mocha:cli:run:helpers');
+const Mocha = require('../mocha');
+const path = require('path');
+const utils = require('../utils');
+const minimatch = require('minimatch');
+const chalk = require('chalk');
+const symbols = require('log-symbols');
+
+const cwd = (exports.cwd = process.cwd());
+
+/**
+ * Exits Mocha when tests + code under test has finished execution (default)
+ * @param {number} code - Exit code; typically # of failures
+ * @private
+ */
+const exitLater = code => {
+  process.on('exit', () => {
+    process.exit(Math.min(code, 255));
+  });
+};
+
+/**
+ * Exits Mocha when Mocha itself has finished execution, regardless of
+ * what the tests or code under test is doing.
+ * @param {number} code - Exit code; typically # of failures
+ * @private
+ */
+const exit = code => {
+  const clampedCode = Math.min(code, 255);
+  let draining = 0;
+
+  // Eagerly set the process's exit code in case stream.write doesn't
+  // execute its callback before the process terminates.
+  process.exitCode = clampedCode;
+
+  // flush output for Node.js Windows pipe bug
+  // https://github.com/joyent/node/issues/6247 is just one bug example
+  // https://github.com/visionmedia/mocha/issues/333 has a good discussion
+  const done = () => {
+    if (!draining--) {
+      process.exit(clampedCode);
+    }
+  };
+
+  const streams = [process.stdout, process.stderr];
+
+  streams.forEach(stream => {
+    // submit empty write request and wait for completion
+    draining += 1;
+    stream.write('', done);
+  });
+
+  done();
+};
+
+/**
+ * Hide the cursor.
+ * @private
+ */
+const hideCursor = () => {
+  process.stdout.write('\u001b[?25l');
+};
+
+/**
+ * Show the cursor.
+ * @private
+ */
+const showCursor = () => {
+  process.stdout.write('\u001b[?25h');
+};
+
+/**
+ * Stop cursor business
+ * @private
+ */
+const stop = () => {
+  process.stdout.write('\u001b[2K');
+};
```

Comment:
Shouldn't we create a cursor handling module (more in `Base.cursor`) and reuse it?

Reply:
browser doesn't need it.

Reply:
True, but both CLI and Base reporter do... they could share it, and stub it in "browserify"

Reply:
feel free to make refactors... might be reasonable to extract the fiddly cursor stuff, or pull a module down from npm that does it better.  I'm going to pass for this PR though